### PR TITLE
clarity-city: init at 1.0.0

### PR DESCRIPTION
**Pull request by sagikazarmark**
## Description of changes

Clarity City is a font used by the CNCF

https://github.com/vmware/clarity-city

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by donovanglover** [comment]
Result of `nixpkgs-review pr 258967` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>clarity-city</li>
  </ul>
</details>

File: pkgs/data/fonts/clarity-city/default.nix [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+stdenvNoCC.mkDerivation rec {
+  pname = "clarity-city";
+  version = "1.0.0";
+
+  src = fetchFromGitHub {
+    owner = "vmware";
+    repo = "clarity-city";
+    rev = "v${version}";
+    sha256 = "sha256-1POSdd2ICnyNNmGadIujezNK8qvARD0kkLR4yWjs5kA=";
```
Comment (donovanglover) [inline pkgs/data/fonts/clarity-city/default.nix:11]:
```suggestion
    hash = "sha256-1POSdd2ICnyNNmGadIujezNK8qvARD0kkLR4yWjs5kA=";
```

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -29309,6 +29309,8 @@ with pkgs;
 
   chunk = callPackage ../data/fonts/chunk { };
 
+  clarity-city = callPackage ../data/fonts/clarity-city { };
```
Comment (donovanglover) [inline pkgs/top-level/all-packages.nix:29312]: Use the new [`pkgs/by-name`](https://github.com/NixOS/nixpkgs/tree/master/pkgs/by-name) format so you don't have to edit this file

File: pkgs/data/fonts/clarity-city/default.nix [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    install -Dm644 OpenType/*.otf -t $out/share/fonts/opentype
+    install -Dm644 Webfonts/EOT/*.eot -t $out/share/fonts/eot
+    install -Dm644 Webfonts/WOFF/*.woff -t $out/share/fonts/woff
+    install -Dm644 Webfonts/WOFF2/*.woff2 -t $out/share/fonts/woff2
+
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    description = "Clarity City is an open source sans-serif typeface";
```
Comment (donovanglover) [inline pkgs/data/fonts/clarity-city/default.nix:27]: Descriptions [must not](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#meta-attributes) start with the package name

**Review by donovanglover — CHANGES_REQUESTED** [review]

**Review by donovanglover — CHANGES_REQUESTED** [review]

**Comment by sagikazarmark** [comment]
@donovanglover thank you for the review! I made the requested changes.

**Review by donovanglover — COMMENTED** [review]
LGTM after squashing the commits :+1:

**Comment by sagikazarmark** [comment]
Squash done

**Review by donovanglover — APPROVED** [review]
LGTM

**Pull request merged**